superset-frontend/src/dashboard/components/Header.jsx

```diff
@@ -318,11 +345,17 @@ class Header extends React.PureComponent {
       hasUnsavedChanges,
       isLoading,
       refreshFrequency,
+      isPersistentRefreshFrequency,
       setRefreshFrequency,
     } = this.props;
 
     const userCanEdit = dashboardInfo.dash_edit_perm;
     const userCanSaveAs = dashboardInfo.dash_save_perm;
+    const refreshLimit =
+      dashboardInfo?.common?.conf?.SUPERSET_DASHBOARD_PERIODICAL_REFRESH_LIMIT;
```

Comment:
why do we need the optional chaining here? I'd expect the `conf` object always to be included in dashboardInfo since I believe it comes down with the initial bootstrap payload.

Reply:
i feel handle unexpected null exception is always useful?

Reply:
I'd disagree. If we know something won't be null or undefined, then we shouldn't try to handle that case. Note that if this was typed, then you'd get an error for unnecessary optional chaining

Reply:
Ok. I removed optional chaining for a few places (all dashboardInfo related).